### ISSUE TITLE
Slightly jittery animation after click (WebGL)

### ISSUE THREAD

**Issue by tobireif** [GoodBoyDigital/pixi.js#1210]
The following issue occurs sometimes, in Firefox and Chrome and Safari, but not always (eg it occurs after each click, but not on a different day). I hope you can you can repoduce it.

I also submitted the report to Apple as a Mac OS issue (and to Firefox and Chrome), but I thought perhaps there's something that you guys can improve, or perhaps you can request required changes from the guys implementing the underlying layers (eg browser/OS/driver).

When I load
http://tobireif.com/boomchat_copy_2_for_bugreport/web/
in Firefox/Chrome/Safari, and click on one of the emoticons, the then following animation stutters slightly for the first ~0.7 seconds, then becomes smooth and fast as expected.

For testing: When you click the emoticons repeatedly and quickly, the triggers get queued and the animations start in sequence without requiring current clicks. These animations are all smooth. But when you wait for eg the first animation to finish completely and only then click again, the first and second animations are jittery (and each following animation that has been started with a click after waiting for the previous animation to fully finish). Perhaps the reaction to the click slows the animation down?

Please compare the click-triggered animations with the queue-triggered animations - the issue is barely noticable but still crucial for me. I hope your hardware/software combo allows you to repoduce it.

Info: MacBook Pro, NVIDIA GeForce 9400M 256 MB, Mac OS X 10.9.5.

If you can't reproduce it (using the linked page), feel free to close this issue.

**Comment by GoodBoyDigital** [comment]
Hello! is this still a thing? can we close? cheers!

**Comment by tobireif** [comment]
Hi Mat!

Recently I found out that the issue occurs when the MacBook is not plugged in / runs on battery.
Apple says they've fixed it in Yosemite. Will try to test on Yosemite as soon as I can.

But since Apple is working on it / fixed it, you can close this ticket here (regarding that issue). But the listed functions could perhaps still be improved, eg this:
  ```
  "TODO optimize by not check EVERY TIME! maybe half as often?"
  ```
and these: InteractionManager.onMouseMove (100%), InteractionManager.onMouseDown (93%), InteractionManager.hitTest (94%), Text.updateText (72%), WebGLRenderer.render (98%). Chad had added the labels "enhancement" and "Interaction Manager", and had added this ticket to the v3.0 milestone, so I don't now whether it can be closed.

There still are some perf issues in my web app which might be Pixi related, I'll report them in a separate ticket.

**Comment by GoodBoyDigital** [comment]
Thanks Tobi, have you tried profiling in chrome? Those results seem insanely slow? Or maybe I don't fully understand what the results mean..

**Comment by englercj** [comment]
Yeah when I profile this, I get 94% idle.

**Comment by tobireif** [comment]
Again, the percentages are each from a low-framerate phase.

Also, there is a severe issue when this MacBook here runs on battery, that's where the results come from. Apple is working on fixing that issue.

Checked in Chrome: There also are some drastic framerate drops, sometimes involving InteractionManager.onMouseUp .

My web app changed a lot (I stopped using Box2D). If I find some perf issues which might be Pixi related, I'll report them in a separate ticket (because the main issue of this ticket here is caused by the OS or eg drivers and gets handled by Apple).

Unless there's something to improve regarding the listed InteractionManager functions (and the TODO item from the source :) ) feel free to close this ticket.
